packages/react-components/src/types/Attachment.ts

```diff
@@ -44,48 +83,69 @@ export interface AttachmentMenuAction {
 
 /**
  * @beta
- * A attachment download error returned via a {@link FileDownloadHandler}.
- * This error message is used to render an error message in the UI.
  */
-export interface FileDownloadError {
-  /** The error message to display in the UI */
-  errorMessage: string;
+export interface AttachmentUploadOptions {
+  /**
+   * A string containing the comma separated list of accepted file types.
+   * Similar to the `accept` attribute of the `<input type="file" />` element.
+   * Accepts any type of file if not specified.
+   * @beta
+   */
+  acceptedMimeTypes?: string[];
+  /**
+   * Allows multiple files to be selected if set to `true`.
+   * Similar to the `multiple` attribute of the `<input type="file" />` element.
+   * @defaultValue false
+   * @beta
+   */
+  canUploadMultiple?: boolean;
+  /**
+   * A function of type {@link AttachmentUploadHandler} for handling file uploads.
+   * @beta
+   */
+  handler: AttachmentUploadHandler;
+}
+
+/**
+ * A wrapper object for a file that is being uploaded.
+ * Allows managing file uploads by providing common functions for updating the
+ * upload progress, canceling an upload, completing an upload etc.
+ * @beta
+ */
+export interface AttachmentUploadManager {
+  /**
+   * Unique identifier for the file upload.
+   */
+  id: string;
+  /**
+   * HTML {@link File} object for the uploaded file.
+   */
+  file?: File;
+  /**
+   * Update the progress of the upload.
+   * @param value - number between 0 and 1
+   */
+  notifyUploadProgressChanged: (value: number) => void;
+  /**
+   * Mark the upload as complete.
+   * Requires the `metadata` param containing uploaded file information.
+   * @param metadata - {@link AttachmentMetadata}
+   */
+  notifyUploadCompleted: (metadata: AttachmentMetadata) => void;
+  /**
+   * Mark the upload as failed.
+   * @param message - An error message that can be displayed to the user.
+   */
+  notifyUploadFailed: (message: string) => void;
```

Comment:
Should we return the metadata as a secondary parameter to be able to know which failed? `notifyUploadCompleted` passes the metadata back, so does it make sense to be consistent?

Reply:
no this call back is specific to each file instance

for example:

```
file1.notifyUploadFailed("failed")
file2.notifyUploadFailed("failed")
```

Reply:
so why is this different from the `notifyUploadCompleted` which passes the metadata?

Reply:
i mean the file passed by upload manager is different from attachment required by notifyUploadCompleted right


notifyUploadCompleted takes in an attachment metadata and basically to convert file -> attachment
while notifyUploadFailed is to convert file -> attachment and set uploadStatus.message to the one provided from user